File: Algorithms/Two Pointer/11. Container With Most Water.py
class Solution(object):
	def maxArea(self, height):
		left = 0
		right = len(height) - 1
		ans = 0
		while(left < right):
			L = right - left
			if(height[left] < height[right]):
				area = height[left] * L
				left += 1
			else:
				area = height[right] * L
				right -= 1
			ans = area if area > ans else ans
		
		return ans

File: Algorithms/Two Pointer/11. Container With Most Water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
		int left = 0, right = height.size()-1;
		int ans = 0;
		int L, area;
		while(left < right) {
			L = right - left;	
			if(height[left] < height[right]) {
				area = height[left] * L;
				left += 1;
			}
			else {
				area = height[right] * L;
				right -= 1;
			}
			if(ans < area) ans = area;
		}
       return ans; 
    }
};
